src/core/PathMaster.ts

```diff
@@ -224,33 +226,49 @@ export class PathMaster {
         return "";
     }
 
+    private tryReadPkg (directory) {
+        try {
+            return require(path.join(directory, "package.json"))
+        } catch(e) {
+            void e
+        }
+    }
+
     private testFolder(folder: string, name: string) {
-        const extensions = ["js", "jsx"];
+        let extensions = jsExtensions;
         if (this.tsMode) {
-            extensions.push("ts", "tsx");
+            extensions = tsExtensions
         }
 
         if (fs.existsSync(folder)) {
-            for (let i = 0; i < extensions.length; i++) {
-                let ext = extensions[i];
-                const index = `index.${ext}`;
-                const target = path.join(folder, index);
-                if (fs.existsSync(target)) {
-                    let result = path.join(name, index);
-                    let startsWithDot = result[0] === "."; // After transformation we need to bring the dot back
-                    if (startsWithDot) {
-                        result = `./${result}`;
-                    }
-                    return result;
+            let result
+            const pkg = this.tryReadPkg(folder)
+            if (pkg && pkg.main) {
+                result = path.join(name, pkg.main)
```

Comment:
This should return an alias (FuseBox will regenerate the code and point to a correct file)